strider/trapi.py

```diff
@@ -410,6 +414,92 @@ async def fill_categories_predicates(
                 node["categories"] = []
 
 
+def convert_subclasses_to_aux_graphs(
+    message: Message,
+    kp_id: str,
+    logger: logging.Logger = logging.getLogger(),
+):
+    """Take any subclass results and move them to aux graphs."""
+    for result in message.results:
+        for node_binding in result.node_bindings.values():
+            for node in node_binding:
+                if node.query_id is not None:
+                    # move query_id to id
+                    subclass = node.id
+                    node.id = node.query_id
+                    node.query_id = None
+
+                    primary_edge_id = str(uuid.uuid4())[:8]
+                    aux_edge_id = str(uuid.uuid4())[:8]
+                    subclass_edge_id = str(uuid.uuid4())[:8]
+                    original_edge_id = None
+                    original_edge = None
+
+                    # get original edge
+                    for analysis in result.analyses:
+                        for edge_bindings in analysis.edge_bindings.values():
+                            for edge_binding in edge_bindings:
+                                original_edge_id = edge_binding.id
+                                original_edge = message.knowledge_graph.edges[
+                                    original_edge_id
+                                ]
+                                # set result edge binding as one with superclass
+                                edge_binding.id = primary_edge_id
+
+                    # create subclass edge in knowledge graph
+                    message.knowledge_graph.edges[subclass_edge_id] = Edge.parse_obj(
+                        {
+                            "subject": subclass,
+                            "object": node.id,
+                            "predicate": "biolink:subclass_of",
+                            "attributes": [],
+                            "sources": [
+                                {
+                                    "resource_id": kp_id,
+                                    "resource_role": "primary_knowledge_source",
+                                },
+                            ],
+                        }
+                    )
+
+                    # create primary edge with support graph
+                    message.knowledge_graph.edges[primary_edge_id] = copy.deepcopy(
+                        original_edge
+                    )
+                    if original_edge.subject == subclass:
+                        message.knowledge_graph.edges[primary_edge_id].subject = node.id
+                    else:
+                        message.knowledge_graph.edges[primary_edge_id].object = node.id
+
+                    had_support_graphs = False
+                    for attribute in message.knowledge_graph.edges[
+                        primary_edge_id
+                    ].attributes:
+                        if attribute.attribute_type_id == "biolink:support_graphs":
```

Comment:
I don't think we should do this since the attribute_source would be incorrect.

Reply:
I'm not sure I understand. This code block is adding the aux graph edge to the the support_graphs attribute of the new edge that has the superclass curie in it. This edge is still being returned by the KP, just in a different format.

Reply:
Maybe I'm misunderstanding, but I was under the impression that we were checking all of the attributes, and if there already existed an attribute with the "biolink:support_graphs" as the attribute_type_id, then we add the aux graph id to that attribute. If there are no attributes like that, then we create a new attribute, correct?

There should be another "attribute_source" field on that attribute. That field should have "infores:aragorn" listed as the source, but that won't happen in this case if the kp already returned with an attribute that was "biolink:support_graphs".

Reply:
Added that in.